test/inductor/test_inplace_padding.py

```diff
@@ -0,0 +1,168 @@
+    enable_inplace_padding = os.environ.get("TORCHINDUCTOR_INPLACE_PADDING") == "1"
+
+DO_PERF_TEST = os.environ.get("DO_PERF_TEST") == "1"
+
+
+@inductor_config.patch(inplace_padding=enable_inplace_padding)
+class InplacePaddingTest(TestCase):
+    def test_pad_nonzero_fusible(self):
+        """
+        If the padding can be fused with downstream op, there would
+        be little benefit to do inplace padding.
+        """
+
+        def f(x):
+            x = aten.constant_pad_nd(x, (0, 8, 0, 0), 12345.0)
+            return x.sum(dim=-1)
+
+        M, N = 2048, 2048
+        x = rand_strided((M, N), (N + 10, 1), device=GPU_TYPE)
+        check_model(self, f, (x,), atol=1e-3, rtol=1e-3)
+
+        self.assertEqual(num_inplace_padding(), 0)
+
+    def test_pad_nonzero_non_fusible(self):
+        def f(x, y):
+            x = aten.constant_pad_nd(x, (0, 8, 0, 0), 12345.0)
+            return x @ y
+
+        M, N = 2048, 2048
+        x = rand_strided((M, N), (N + 10, 1), device=GPU_TYPE)
+        y = torch.randn(N + 8, M, device=GPU_TYPE)
+        check_model(self, f, (x, y), atol=1e-2, rtol=1e-2)
+
+        self.assertEqual(num_inplace_padding(), 1)
+
+    def test_pad_too_large(self):
+        def f(x, y):
+            x = aten.constant_pad_nd(x, (0, 8, 0, 0), 12345.0)
+            return x @ y
+
+        M, N = 2048, 2048
+        x = rand_strided((M, N), (N + 5, 1), device=GPU_TYPE)
+        y = torch.randn(N + 8, M, device=GPU_TYPE)
+        check_model(self, f, (x, y), atol=1e-2, rtol=1e-2)
+
+        self.assertEqual(num_inplace_padding(), 0)
+
+    def test_mutating_padding_input(self):
+        """
+        Even if `aten.constant_pad_nd` input get inplace updated,
+        doing inplace-padding still generates the correct result.
+        """
+
+        def f(x, y):
+            x2 = aten.constant_pad_nd(x, (0, 8, 0, 0), 12345.0)
+            x.add_(5)
+            return x2 @ y
+
+        M, N = 2048, 2048
+        x = rand_strided((M, N), (N + 10, 1), device=GPU_TYPE)
+        y = torch.randn(N + 8, M, device=GPU_TYPE)
+        check_model(self, f, (x, y), atol=1e-2, rtol=1e-2)
+
+        self.assertEqual(num_inplace_padding(), 1)
+
+    def test_mutating_padding_output(self):
+        """
+        Inplace padding does not take effect since the `aten.add_` op
+        cause the user of the padding output to be not matmul. We skip
+        inplace-padding in this case.
+        """
+
+        def f(x, y):
+            x = aten.constant_pad_nd(x, (0, 8, 0, 0), 12345.0)
+            x.add_(1)
+            return x @ y
+
+        M, N = 2048, 2048
+        x = rand_strided((M, N), (N + 10, 1), device=GPU_TYPE)
+        y = torch.randn(N + 8, M, device=GPU_TYPE)
+        # 1e-3 tolerance may fail on CI A10G GPU.
+        check_model(self, f, (x, y), atol=1e-2, rtol=1e-2)
+
+        self.assertEqual(num_inplace_padding(), 0)
+
+    @unittest.skipIf(
+        torch.cuda.is_available()
```

Comment:
Can you resolve the issue #143479 here? Since there is no `CUBLAS_WORKSPACE_CONFIG` on other GPU.

Reply:
I've resolve this in a local commit. Will update after resolve a few remaining tiny things for fixing the failed internal model reported.

Reply:
Thanks^^